tests/functional/testServerSearch.js

```diff
@@ -0,0 +1,188 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const assert = require("assert");
+const queries = require("./queries.js");
+const vpn = require("./helper.js");
+
+describe("Server list", function () {
+  let servers;
+  let currentCountryCode;
+  let currentCity;
+
+  this.timeout(240000);
+  this.ctx.authenticationNeeded = true;
+
+  beforeEach(async () => {
+    await vpn.waitForQueryAndClick(queries.screenHome.SERVER_LIST_BUTTON);
+    await vpn.waitForQuery(queries.screenHome.STACKVIEW.ready());
+
+    servers = await vpn.servers();
+    currentCountryCode = await vpn.getVPNProperty(
+      "VPNCurrentServer",
+      "exitCountryCode"
+    );
+    currentCity = await vpn.getVPNProperty("VPNCurrentServer", "exitCityName");
+
+    for (let server of servers) {
+      if (currentCountryCode === server.code) {
+        for (let city of server.cities) {
+          if (city.name == currentCity) {
+            currentCity = city.localizedName;
+            break;
+          }
+        }
+      }
+    }
+    console.log(
+      "Current city (localized):",
+      currentCity,
+      "| Current country code:",
+      currentCountryCode
+    );
+  });
+
+  it.only("multihop search", async () => {
+    await vpn.waitForQueryAndClick(
+      queries.screenHome.serverListView.MULTIHOP_SELECTOR_TAB.visible()
+    );
+    await vpn.waitForQuery(
+      queries.screenHome.serverListView.ENTRY_BUTTON
+    );
+    await vpn.waitForQueryAndClick(
+      queries.screenHome.serverListView.EXIT_BUTTON.visible()
+    );
+
+    const server = servers[0];
+    console.log("server name: ", server.name);
+    await vpn.waitAndSendkeys(
+      queries.screenHome.serverListView.SEARCH_BAR_TEXT_FIELD.visible(),
+      server.name
+    );
+
+    const countryId = queries.screenHome.serverListView.generateCountryId(
+      server.code
+    );
+    await vpn.waitForQuery(countryId.visible());
```

Comment:
I think here you want to run a different query to retrieve the name of the first server item. And see if the text matches with the selected country.
Plus don't use server[0]. Use the last server to make things more complex.